extension/src/popup/views/__tests__/Swap.test.tsx

```diff
@@ -0,0 +1,310 @@
+
+jest.mock("popup/helpers/horizonGetBestPath", () => ({
+  get horizonGetBestPath() {
+    return jest.fn(() => ({
+      path: [
+        {
+          asset_code: "TEST",
+          asset_issuer:
+            "GCQQKT67XY6N2GTAH3D2Q3AGKYYC6TD33AL2Y36HYT6PKI2SLWDHAYYM",
+          asset_type: "credit_alphanum4",
+        },
+      ],
+      source_amount: "20",
+      source_asset_type: "credit_alphanum4",
+      source_asset_code: "XLM",
+      source_asset_issuer: "native",
+      destination_amount: "10",
+      destination_asset_type: "credit_alphanum4",
+      destination_asset_code: "USDC",
+      destination_asset_issuer:
+        "GCK3D3V2XNLLKRFGFFFDEJXA4O2J4X36HET2FE446AV3M4U7DPHO3PEM",
+    }));
+  },
+}));
+
+jest.spyOn(ApiInternal, "signFreighterTransaction").mockImplementation(() =>
+  Promise.resolve({
+    signedTransaction:
+      "AAAAAgAAAADaBSz5rQFDZHNdV8//w/Yiy11vE1ZxGJ8QD8j7HUtNEwAAAGQAAAAAAAAAAQAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAQAAAADaBSz5rQFDZHNdV8//w/Yiy11vE1ZxGJ8QD8j7HUtNEwAAAAAAAAAAAvrwgAAAAAAAAAABHUtNEwAAAEBY/jSiXJNsA2NpiXrOi6Ll6RiIY7v8QZEEZviM8HmmzeI4FBP9wGZm7YMorQue+DK9KI5BEXDt3hi0VOA9gD8A",
+  }),
+);
+
+jest
+  .spyOn(ApiInternal, "submitFreighterTransaction")
+  .mockImplementation(() => Promise.resolve({}));
+
+jest.spyOn(UseNetworkFees, "useNetworkFees").mockImplementation(() => ({
+  recommendedFee: "0.00001",
+  networkCongestion: UseNetworkFees.NetworkCongestion.MEDIUM,
+}));
+
+const mockHistoryGetter = jest.fn();
+jest.mock("popup/constants/history", () => ({
+  get history() {
+    return mockHistoryGetter();
+  },
+}));
+
+const publicKey = "GCXRLIZUQNZ3YYJDGX6Z445P7FG5WXT7UILBO5CFIYYM7Z7YTIOELC6O";
+
+describe("Swap", () => {
+  beforeEach(async () => {
+    const history = createMemoryHistory();
+    history.push(ROUTES.swap);
+    mockHistoryGetter.mockReturnValue(history);
+
+    render(
+      <Wrapper
+        history={history}
+        state={{
+          auth: {
+            error: null,
+            applicationState: ApplicationState.PASSWORD_CREATED,
+            publicKey,
+            allAccounts: mockAccounts,
+            hasPrivateKey: true,
+          },
+          settings: {
+            networkDetails: TESTNET_NETWORK_DETAILS,
+            networksList: DEFAULT_NETWORKS,
+          },
+        }}
+      >
+        <Swap />
+      </Wrapper>,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("send-amount-view")).toBeDefined();
+    });
+  });
+
+  afterAll(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders swap view initial state", () => {
+    expect(screen.getByTestId("AppHeaderPageTitle")).toHaveTextContent(
+      "Swap XLM",
+    );
+    expect(screen.getByTestId("AppHeaderPageSubtitle")).toHaveTextContent(
+      "220.49999 XLM available",
+    );
+    expect(screen.getByTestId("send-amount-amount-input")).toHaveValue("0");
+
+    const assetSelects = screen.getAllByTestId("AssetSelect");
```

Comment:
I noticed we're using two conventions for test IDs: camelcase and kebab case. I used camelcase, but I can change it to kebab if we prefer that.

Reply:
Oh yeah, this isn't consistent in the repo 😬 I think camel is fine!